pkg/infra/httpclient/httpclientprovider/forwarded_cookies_middleware.go

```diff
@@ -0,0 +1,24 @@
+package httpclientprovider
+
+import (
+	"net/http"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend/httpclient"
+	"github.com/grafana/grafana/pkg/util/proxyutil"
+)
+
+const ForwardedCookiesMiddlewareName = "forwarded-cookies"
+
+// ForwardedCookiesMiddleware middleware that sets Cookie header on the
+// outgoing request, if forwarded cookies configured/provided.
+func ForwardedCookiesMiddleware(forwardedCookies []*http.Cookie, allowedCookies []string) httpclient.Middleware {
```

Comment:
Feels like needing to do the filtering here mixes concerns. Should be easy to just pass on cookies to forward and have a util function to do the filtering before in the caller code.

Reply:
hmm and can we change it anyway? This signature seems to me like `addNumberButFilterOutOneNumber(numbers []int, withoutNumber int)`. Like it seems like it should be much easier to just compose it like `addNumbers(numbers.filter(numberToFilter))` or something like that.

Reply:
@aocenas oh yeah, for sure. it can be changed if you have ideas for improvements. i'd recommend doing that in a separate PR so it's easier to get feedback from `@grafana/plugins-platform-backend` without the prometheus changes mixed in